src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private static final double DEFAULT_SIZE_GROW = 1.5;
+
+    private int size;
+    private T[] values;
+
+    public ArrayList() {
+        values = (T[]) new Object[MAX_ITEMS_NUMBER];
+    }
+
+    private T[] growSize() {
+        int newCapacity = 0;
+        int oldCapacity = values.length;
+        newCapacity = (int) (oldCapacity * DEFAULT_SIZE_GROW);
+
+        T[] newValues = (T[]) new Object[newCapacity];
+        System.arraycopy(values, 0, newValues, 0, size);
+        return newValues;
+    }
+
     @Override
     public void add(T value) {
+        if (size == values.length) {
+            values = growSize();
+        }
+        values[size] = value;
+        size++;
 
     }
 
     @Override
-    public void add(T value, int index) {
+    public void add(T value, int index) throws ArrayListIndexOutOfBoundsException {
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index");
+        }
+        if (size == values.length) {
+            values = growSize();
+        }
+        T[] resultCopy = (T[]) new Object[values.length];
+        System.arraycopy(values, 0, resultCopy, 0, index + 1);
+        resultCopy[index] = value;
+        System.arraycopy(values, index, resultCopy, index + 1, size - index);
+        values = resultCopy;
+        size++;
 
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
-    public T get(int index) {
-        return null;
+    public T get(int index) throws ArrayListIndexOutOfBoundsException {
+        if (index >= size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Wrong index");
+        } else {
+            return values[index];
+        }
+
     }
 
     @Override
-    public void set(T value, int index) {
+    public void set(T value, int index) throws ArrayListIndexOutOfBoundsException {
+        if (index >= size || index < 0) {
```

Comment:
you have similar checks nearly in every method, please extract them to separate method(s) and reuse

Reply:
not fixed, you have similar code on lines 63-65, 73-75, 82-84
please introduce some void method like checkIndex or something like that that will contain this logic